web/models/project.ex

```diff
@@ -0,0 +1,68 @@
+defmodule CodeCorps.Project do
+  @moduledoc """
+  Represents a project on Code Corps, e.g. the "Code Corps" project itself.
+  """
+
+  use CodeCorps.Web, :model
+
+  import CodeCorps.Validators.SlugValidator
+
+  schema "projects" do
+    field :description, :string
+    field :icon_large_url, :string
+    field :icon_thumb_url, :string
+    field :long_description_body, :string
+    field :long_description_markdown, :string
+    field :slug, :string
+    field :title, :string
+
+    belongs_to :organization, CodeCorps.Organization
+
+    timestamps()
+  end
+
+  def changeset(struct, params) do
+    struct
+  	|> cast(params, [:title, :description, :long_description_markdown])
+    |> validate_required(:title)
+    |> generate_slug()
+    |> validate_slug(:slug)
+    |> add_project_icons(Map.get(params, :base64_icon_data))
+    |> render_markdown_to_html
+  end
+
+  defp add_project_icons(changeset, nil), do: changeset
+  defp add_project_icons(changeset, base64_icon_data) do
+    # TODO: Deal with base64 conversion to image file
+    # TODO: Deal with image file upload
+    changeset
+  end
+
+  defp generate_slug(changeset) do
+    slug =
+      get_change(changeset, :title)
+      |> do_generate_slug
+
+    changeset
+    |> put_change(:slug, slug)
+  end
+  defp do_generate_slug(nil), do: nil
+  defp do_generate_slug(value), do: Inflex.parameterize(value)
```

Comment:
Didn't realise we now have the function in a separate module. Switched to using that now.